### PR TITLE
Add download retry tests

### PR DESCRIPTION
**Pull request by GregorioDiStefano**
Signed-off-by: Greg Di Stefano <greg.distefano+github@gmail.com>

### CODE REVIEW

**Comment by GregorioDiStefano** [comment]
@kacf please take a look

**Comment by GregorioDiStefano** [comment]
Need to clean this up a bit more before review

**Comment by GregorioDiStefano** [comment]
cleaned up, please take a look now :)

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
+  run.sh --runslow ); fi
+- if [ "$JOB_TYPE" = compile_and_integration_tests_nightly ]; then ( cd tests && bash
+  run.sh --runnightly ); fi
 
-    - if [ "$JOB_TYPE" = compile_and_integration_tests_slow ]; then
-        ( cd tests && bash run.sh --runslow );
-      fi
+notifications:
+  slack:
+    secure: l831fv7p5iskdhG082XVdi5UBhYXEZyRn0ZgEfPWqLfqCj5FyxC36KaYiQo+vp8BS9vX0JFEOuZ4PiKjG3+jSEsJIk0Lx/1fjCg1zi13xfzI2ke1BWGuBT5tuz+nKj3TujfnHM0KE8Ix8GkwnamR388DO0FtgJrhEMBylo33zVV5ZESBa3B2bwUfWgfFF7VeUeqtDnOiHFDqsu2A4lOmMMJPp8k+71x0qvUsUStswrG11S0UhDReuuH1+siy3/bDEnrZIwEBvSM8DxdLLWC/SmCiCMdCU95GfzFRJsZN+2+KavCN8kXxvOG8rYrr8EhOLF4BbGDTJI9Dw+4H4OKOD/Bf+hr1OqYxpGtJxKzGWVlds1MwjjxklWlXopgIfwIZtmgC2RynLEw7LYHMcmBmr2+zjbvl0LZqWto1/G+la3J45rNaXawJtwY2RJ+c5jkm13BoYbOpUAg2SZ7WwW6frW6XdObiegpBfObSapAIEwMHeCSYvZbDPfhv0PFgGS9BQwJ+bz0PY1xoY45EaglnLt8zhcG3pfzXMo7jHFwIcP9/NGi7SO71lL4uhmcYn29PhRoIBMg/gjq7doycCfBWyUQrjGBOhDHzpq01TazxakDussWCSQ534ZwWrksI1d9ONMyikNOFXBkSHsqihYrNF+sLAtGEtHg0Ww7qRc0P/R0=
```
Comment (kacf) [inline .travis.yml]: This breaks with the indentation style in all our other `.travis.yml` files. I would stick to the existing way, unless we intend to make this our new style.

File: tests/tests/test_fault_tolerance.py [hunk 2]
```diff
@@ -26,6 +26,27 @@
...
+                if int(output) >= 4:  # check that some retries have occured
+                    logging.info("Looks like the download was retried 2 times, restoring download functionality")
+                    break
+
+        if timeout_time <= int(time.time()):
+            pytest.fail("timed out waiting for download retries")
+
+        # make sure that retries happen after 5 minutes have passed
+        assert timeout_time - start_time >= 5 * 60, "Ooops, looks like the retry happend within less than 5 minutes"
+        logging.info("Waiting for system to finish download")
```
Comment (kacf) [inline tests/tests/test_fault_tolerance.py]: Holy crap, five min! This is why we really need to get parallel tests working.

File: .travis.yml (hunk 1 above)
Comment (GregorioDiStefano) [inline .travis.yml]: I used the travis cmdline tool to add the encrypted string. It destroyed the formatting of the yaml file and I didn't notice. Fixed now

File: tests/tests/test_fault_tolerance.py (hunk 2 above)
Comment (GregorioDiStefano) [inline tests/tests/test_fault_tolerance.py]: yup

File: .travis.yml (hunk 1 above)
Comment (kacf) [inline .travis.yml]: Sorry, but this is still the wrong style. You're changing several lines that are already correct. The correct indentation (or at least the one we use) is four spaces per level.

Sorry to be pedantic about this, but stuff like this quickly adds up and makes our files less readable.

**Comment by GregorioDiStefano** [comment]
fixed

**Pull request merged**